apps/synapse-portal-framework/src/components/HeaderSearchBox.tsx

```diff
@@ -42,17 +45,19 @@ const HeaderSearchBox = ({
 
   const handleTermClick = (term: string) => {
     const trimmedTerm = term.trim()
-    setSearchParams(prev => {
-      prev.set(FTS_SEARCH_TERM, trimmedTerm)
-      if (role) {
-        prev.set(FTS_SEARCH_ROLE, role)
-      }
-      return prev
-    })
-
     if (path) {
+      setSearchParams(prev => {
+        prev.set(FTS_SEARCH_TERM, trimmedTerm)
+        if (role) {
+          prev.set(FTS_SEARCH_ROLE, role)
+        }
+        return prev
+      })
       window.location.pathname = `${path}`
```

Comment:
I must have missed this in a previous code review, could you update this to use a react-router hook instead of updating `window` (or create a ticket to follow-up)?